payjoin/src/lib.rs

```diff
@@ -74,3 +74,5 @@ pub mod core;
 pub(crate) mod error_codes;
 #[cfg(feature = "_core")]
 pub use crate::core::error::ImplementationError;
+#[cfg(feature = "_core")]
+pub use crate::core::version::Version;
```

Comment:
I think `pub(crate)` is sufficient since `Version` is an internal API

```suggestion
#[cfg(feature = "_core")]
pub(crate) use crate::core::version::Version;
```

In a follow up, all of these internal crate re-exports can be done inside core/mod.rs, and then something like 

```rs
#[cfg(feature = "_core")]
pub use crate::core::*;
```

can be used in lib.rs to clean up the verbose syntax here. Thanks for starting the cleanup by adding the `core` mod 😎

Reply:
#688